activemq-unit-tests/src/test/java/org/apache/activemq/usecases/MaxUncommittedCountExceededTest.java

```diff
@@ -0,0 +1,217 @@
+        brokerService.deleteAllMessages();
+        assertNotNull(brokerService);
+
+        activemqConnectionFactory = new ActiveMQConnectionFactory(brokerService.getTransportConnectorByScheme("tcp").getPublishableConnectString());
+        connection = activemqConnectionFactory.createConnection();
+        connection.start();
+        session = connection.createSession(true, Session.SESSION_TRANSACTED);
+        methodNameDestinationName = "AMQ.TX." + cleanParameterizedMethodName(testName.getMethodName().toUpperCase());
+        Queue queue = session.createQueue(methodNameDestinationName);
+        messageProducer = session.createProducer(queue);
+        mbeanServer = ManagementFactory.getPlatformMBeanServer();
+        brokerService.getAdminView().addQueue(methodNameDestinationName);
+        queueViewMBean = getQueueViewMBean(new ActiveMQQueue(methodNameDestinationName));
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (messageProducer != null) {
+            try {
+                messageProducer.close();
+            } catch (Exception e) {
+            } finally {
+                messageProducer = null;
+            }
+        }
+
+        if (session != null) {
+            try {
+                session.close();
+            } catch (Exception e) {
+            } finally {
+                session = null;
+            }
+        }
+
+        if (connection != null) {
+            try {
+                connection.close();
+            } catch (Exception e) {
+            } finally {
+                connection = null;
+            }
+        }
+
+        methodNameDestinationName = null;
+        activemqConnectionFactory = null;
+        if(brokerService != null) {
+            brokerService.deleteAllMessages();
+            brokerService.stop();
+            brokerService.waitUntilStopped();
+        }
+    }
+
+    protected static String cleanParameterizedMethodName(String methodName) {
+        // clean up parameterized method string:
+        // TESTMESSAGETIMESTAMPTIMETOLIVE[DESTINATIONTYPE=QUEUE, MESSAGETYPE=BYTES]
+        // returns: TESTMESSAGETIMESTAMPTIMETOLIVE.QUEUE.BYTES
+
+        if (methodName == null || (!methodName.contains("[") && !methodName.contains("]"))) {
+            return methodName;
+        }
+
+        String[] step1 = methodName.split("\\[", 2);
+        String[] step2 = step1[1].split("\\]", 2);
+        String[] step3 = step2[0].split(",", 16);
+
+        return step1[0] + "." + step3[0].split("=", 2)[1] + "." + step3[1].split("=", 2)[1];
+    }
+
+    protected QueueViewMBean getQueueViewMBean(ActiveMQDestination destination) throws Exception {
+        return JMX.newMBeanProxy(mbeanServer, BrokerMBeanSupport.createDestinationName(BrokerMBeanSupport.createBrokerObjectName(DEFAULT_JMX_DOMAIN_NAME, DEFAULT_JMX_BROKER_NAME).toString(), destination), QueueViewMBean.class);
+    }
+
+    @Test
+    public void testUncommittedCountExceededSync() throws Exception {
```

Comment:
A lot  of the code in these two tests are identical and copy and pasted. This could be refactored to re-use some of the code or maybe even using parameterized tests to reduce the bloat.

Reply:
Fixed. Test converted to parameterized